osu.Game.Rulesets.Osu/Beatmaps/OsuBeatmapConverter.cs

```diff
@@ -35,7 +35,8 @@ protected override IEnumerable<OsuHitObject> ConvertHitObject(HitObject original
                     RepeatSamples = curveData.RepeatSamples,
                     RepeatCount = curveData.RepeatCount,
                     Position = positionData?.Position ?? Vector2.Zero,
-                    NewCombo = comboData?.NewCombo ?? false
+                    NewCombo = comboData?.NewCombo ?? false,
+                    BaseDifficulty = beatmap.BeatmapInfo.BaseDifficulty

```

Comment:
Rather than assigning `BaseDifficulty` here, we should probably be consuming it in `ApplyDefaults` (where it is already available). Then we don't even need to store `BaseDifficulty` locally, but instead a value for `Preempt` and `HitWindowStrictness` or similar.

Reply:
Going forward I would like to see an implementation similar to osu!mania as well - where it stores `HitWindows` as a local member to which OD is applied to by constructor.

This would go well with the plan to unify hit windows.